row.py

```diff
@@ -988,9 +980,131 @@ def join_spreadsheet_info(filtered_results_file, out_dir):
     logging.info("rumber of rows after exploding ocr results: %i", len(final_exploded_df.index))
 
     #: save results to CSV
-    out_file = out_dir / f"final-ocr-results-{datetime.now().strftime('%Y-%m-%d-%H-%M')}.csv"
+    out_file = out_dir / f"joined-ocr-results-{datetime.now().strftime('%Y-%m-%d-%H-%M')}.csv"
     final_exploded_df.to_csv(out_file)
-    logging.info("saved filtered and joined ocr results to %s", out_file)
+    logging.info("saved cleaned and joined ocr results to %s", out_file)
+
+    return out_dir
+
+
+def filter_results(previous_results_file, out_dir):
+    """filter ocr results down further by filtering out additional irrelevant patterns
+
+    Args:
+        previous_results_file (str): path to the CSV file with cleaned and joined results (path_to_file.csv)
+        out_dir (str): where to save the CSV file results
+
+    Returns:
+        str: the location of the output CSV file with filtered results
+    """
+
+    #: silence pandas SettingWithCopyWarning
+    pd.options.mode.chained_assignment = None
+
+    #: set up variables
+    out_dir = Path(out_dir)
+    out_dir.mkdir(parents=True, exist_ok=True)
+
+    working_df = pd.read_csv(previous_results_file)
+
+    #: ensure the 'text' field is read as a string
+    working_df["text"] = working_df["text"].astype(str)
+
+    #: add a 'keep' field, initialized as 'yes', but be used to mark rows to remove with 'no'
+    working_df["keep"] = "yes"
+
+    #: upper-case all letters
+    working_df["text"] = working_df["text"].str.upper()
+
+    #: remove periods, parenthesis, double colons, and slashes
+    logging.info("Removing periods, parenthesis, double colons, and slashes")
+    working_df["text"] = working_df.apply(lambda r: r["text"].replace(".", "").strip(), axis=1)
```

Comment:
Not really that I've seen.  The colon is mostly OCR'd wrong as a 1 or 8, so in some cases, we accept a 1 or 8 as if it might be a colon in the later regexes.  I have seen some exclamation points at the end of a string because a letter got cutoff by the circle and the OCR thinks it's a '!'.  Some of those are otherwise good, so I'll add a rule to strip trailing exclamation points.